language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/BallerinaTextDocumentService.java

```diff
@@ -455,25 +455,26 @@ void setClientCapabilities(LSClientCapabilities clientCapabilities) {
     @Override
     public CompletableFuture<WorkspaceEdit> rename(RenameParams params) {
         return CompletableFuture.supplyAsync(() -> {
-            WorkspaceEdit workspaceEdit = new WorkspaceEdit();
-
             try {
                 RenameContext context = ContextBuilder.buildRenameContext(params.getTextDocument().getUri(),
                         this.workspaceManager,
                         this.serverContext,
                         params.getPosition());
 
-                Map<String, List<TextEdit>> changes = RenameUtil.rename(context, params.getNewName());
-                workspaceEdit.setChanges(changes);
+                boolean honorsChangeAnnotations = this.clientCapabilities.getTextDocCapabilities()
+                        .getRename().getHonorsChangeAnnotations() != null ?
```

Comment:
Get the following part to a variable
`this.clientCapabilities.getTextDocCapabilities().getRename().getHonorsChangeAnnotations()`